src/MediaReplacer.php

```diff
@@ -0,0 +1,324 @@
+            }
+
+            set_transient('media_replacement_message', 'Media replaced successfully!', 5);
+            $this->purge_cloudflare(wp_get_attachment_url($attachment_id));
+            wp_send_json_success();
+        } catch (\Exception $e) {
+            set_transient('media_replacement_error', $e->getMessage(), 5);
+            return;
+        }
+    }
+
+    /**
+     * Replaces the media file associated with the old attachment ID
+     * with the new file located at the specified path.
+     *
+     * @param int $old_file_id The ID of the old attachment.
+     * @param string $new_file_path The path to the new file.
+     */
+    private function replace_media_file(int $old_file_id, string $new_file_path): bool
+    {
+        try {
+            // Get the old file path
+            $old_file_path = get_attached_file($old_file_id);
+
+            // Check if the old file exists
+            if (file_exists($old_file_path)) {
+                unlink($old_file_path);
+            }
+
+            // Move the new file to the old file's location
+            $file_renamed = rename($new_file_path, $old_file_path);
+
+            // If the file was not renamed, abort
+            if (!$file_renamed) {
+                return false;
+            }
+
+            // Update the attachment metadata with new information
+            $filetype = wp_check_filetype($old_file_path);
+            $attachment_data = array(
+                'ID' => $old_file_id,
+                'post_mime_type' => $filetype['type'],
+                'post_title' => preg_replace('/\.[^.]+$/', '', basename($old_file_path)),
+                'post_content' => '',
+                'post_status' => 'inherit',
+            );
+
+            // Update the database record for the file
+            $post_updated = wp_update_post($attachment_data);
+
+            // If the post was not updated, abort
+            if (is_wp_error($post_updated) || $post_updated === 0) {
+                return false;
+            }
+
+            // Update file metadata
+            // By calling the "wp_update_attachment_metadata" function,
+            // the WP Stateless plugin syncs the file with Google Storage.
+            // https://github.com/udx/wp-stateless/blob/0871da645453240007178f4a5f243ceab6a188ea/lib/classes/class-bootstrap.php#L376
+            $attach_data = wp_generate_attachment_metadata($old_file_id, $old_file_path);
+            $post_meta_updated = wp_update_attachment_metadata($old_file_id, $attach_data);
+
+            // If the post meta was not updated, abort
+            return $post_meta_updated;
+        } catch (\Exception $e) {
+            set_transient('media_replacement_error', $e->getMessage(), 5);
+            return false;
+        }
+    }
+
+    /**
+     * Purge the Cloudflare cache for a specific URL.
+     *
+     * @param string $url The URL to be purged.
+     */
+    private function purge_cloudflare(string $url): void
```

Comment:
Not blocking for this PR, but since this function is quite generic I was wondering whether it should live on a different place in order to be used by other features in the future. But we can also do that when there is a need for that.